### PR TITLE
Document permission requirements for Docker Desktop for Windows

### PR DESCRIPTION
**Pull request by gabriellavengeo**
### Proposed changes

Document permission requirements and privileged helper functionality for Docker Desktop for Windows

### CODE REVIEW

File: desktop/windows/permission-requirements.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+The reason for this approach is that Docker Desktop needs to perform a limited set of privileged operations which are conducted by the privileged helper process `com.docker.service.exe`. This approach allows, following the principle of least privilege, `Administrator` access to be used only for the operations for which it is absolutely necessary, while still being able to use Docker Desktop as an unprivileged user.
+
+## Privileged Helper
+
+The privileged helper `com.docker.service.exe` is a Windows service which runs in the background with `SYSTEM` privileges. It listens on the named pipe `//./pipe/dockerBackendV2`. The developer runs the Docker Desktop application, which connects to the named pipe and sends commands to the service. This named pipe is protected, and only users that are part of the `docker-users` group can have access to it.
+
+Here is a detailed list of the functionalities the service performs:
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:21]:
```suggestion
The service performs the following functionalities: 
```

File: desktop/windows/permission-requirements.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+The reason for this approach is that Docker Desktop needs to perform a limited set of privileged operations which are conducted by the privileged helper process `com.docker.service.exe`. This approach allows, following the principle of least privilege, `Administrator` access to be used only for the operations for which it is absolutely necessary, while still being able to use Docker Desktop as an unprivileged user.
+
+## Privileged Helper
+
+The privileged helper `com.docker.service.exe` is a Windows service which runs in the background with `SYSTEM` privileges. It listens on the named pipe `//./pipe/dockerBackendV2`. The developer runs the Docker Desktop application, which connects to the named pipe and sends commands to the service. This named pipe is protected, and only users that are part of the `docker-users` group can have access to it.
+
+Here is a detailed list of the functionalities the service performs: 
+- Ensuring that `kubernetes.docker.internal` is defined in the Win32 hosts file. Defining the DNS name `kubernetes.docker.internal` allows us to share Kubernetes contexts with containers.
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:22]:
```suggestion
- Ensuring that `kubernetes.docker.internal` is defined in the Win32 hosts file. Defining the DNS name `kubernetes.docker.internal` allows Docker to share Kubernetes contexts with containers.
```

File: desktop/windows/permission-requirements.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+- Securely caching the Registry Access Management policy which is read-only for the developer.
+- Creating the Hyper-V VM `"DockerDesktopVM"` and managing its lifecycle - starting, stopping and destroying it. The VM name is hard coded in the service code so the service cannot be used for creating or manipulating any other VMs.
+- Getting the VHDX disk size.
+- Moving the VHDX file/folder.
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:26]:
```suggestion
- Moving the VHDX file or folder.
```

File: desktop/windows/permission-requirements.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- Starting and stopping the Windows Docker engine and querying whether it is running.
+- Deleting all Windows containers data files.
+- Checking if Hyper-V is enabled.
+- Checking if the bootloader activates Hyper-V.
+- Checking if required Windows features are both installed and enabled.
+- Healthcheck and retrieving the version of the service itself.
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:32]:
```suggestion
- Conducting healthchecks and retrieving the version of the service itself.
```

File: desktop/windows/permission-requirements.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+
+## Containers running as root within the Linux VM
+
+The Linux Docker daemon and containers run in a minimal, special-purpose Linux VM managed by Docker. It is immutable, meaning that users can’t extend it or change the installed software.
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:37]:
```suggestion
The Linux Docker daemon and containers run in a minimal, special-purpose Linux VM managed by Docker. It is immutable, meaning that users can’t extend it or change the installed software. 
```
```suggestion
The Linux Docker daemon and containers run in a minimal, special-purpose Linux VM managed by Docker. It is immutable so users can’t extend it or change the installed software. 
```

File: desktop/windows/permission-requirements.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+- Checking if Hyper-V is enabled.
+- Checking if the bootloader activates Hyper-V.
+- Checking if required Windows features are both installed and enabled.
+- Healthcheck and retrieving the version of the service itself.
+
+
+## Containers running as root within the Linux VM
+
+The Linux Docker daemon and containers run in a minimal, special-purpose Linux VM managed by Docker. It is immutable, meaning that users can’t extend it or change the installed software. 
+This means that although containers run by default as `root`, this does not allow altering the VM and does not grant `Administrator` access to the Windows host machine. The Linux VM serves as a security boundary and limits what resources from the host can be accessed. File sharing uses a user-space crafted file server and any directories from the host bind mounted into Docker containers still retain their original permissions, not giving the user access to any files that it doesn’t already have access to.
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:38]:
```suggestion
This means that although containers run by default as `root`, this does not allow altering the VM and does not grant `Administrator` access to the Windows host machine. The Linux VM serves as a security boundary and limits what resources from the host can be accessed. File sharing uses a user-space crafted file server and any directories from the host bind mounted into Docker containers still retain their original permissions. It does not give the user access to any files that it doesn’t already have access to.
```

File: desktop/windows/permission-requirements.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+## Windows Containers
+
+Unlike the Linux Docker engine and containers which run in a VM, Windows containers are an operating system feature, and run directly on the Windows host with `Administrator` privileges For organizations which would not like their developers to run Windows containers, a `–no-windows-containers` installer flag is available from version 4.11 to disable their use.
```
Comment (aevesdocker) [inline desktop/windows/permission-requirements.md:42]:
```suggestion
Unlike the Linux Docker engine and containers which run in a VM, Windows containers are an operating system feature, and run directly on the Windows host with `Administrator` privileges. For organizations do not want their developers to run Windows containers, a `–no-windows-containers` installer flag is available from version 4.11 to disable their use. 
```

**Review by aevesdocker — COMMENTED** [review]
Thanks for the PR @gabriellavengeo - just a few minor changes needed. 
Also wondering if we need to update 'install from the command line' section on the Windows install page with the additional flags?

File: desktop/windows/permission-requirements.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+title: Docker Desktop permission requirements for Windows
+redirect_from:
+- /desktop/windows/privileged-helper/
+---
+
+This page contains information about the permission requirements for running and installing Docker Desktop on Windows, the functionality of the privileged helper process `com.docker.service.exe` and the reasoning behind this approach, as well as clarification about running containers as `root` as opposed to having `Administrator` access on the host and the privileges of the Windows Docker engine and Windows containers.
+
+## Permission requirements
+
+While Docker Desktop can be run without having `Administrator` privileges, it does require them during installation. On installation the user gets an UAC prompt which allows a privileged helper service to be installed. After that Docker Desktop can be run by users without Administrator privileges, provided that they are members of the `docker-users` group. The user who performs the installation is automatically added to this group, but other users must be added manually. This allows the administrator to control who has access to Docker Desktop.
```
[the remark this answers is not in the document]
Comment (gabriellavengeo) [reply desktop/windows/permission-requirements.md:13]: This is not true on windows however, one gets the admin prompt when running the installer, not on the first run of DD

[the remark this answers is not in the document]
Comment (aevesdocker) [reply desktop/windows/permission-requirements.md:13]: Ah! my mistake

**Comment by gabriellavengeo** [comment]
@aevesdocker  Thanks for the review! I responded to one of the suggestions, fixed the rest. It seems that the `--no-windows-containers` flag is already included in the installer documentation

**Comment by aevesdocker** [comment]
>It seems that the `--no-windows-containers` flag is already included in the installer documentation

*facepalm*

**Review by aevesdocker — APPROVED** [review]

**Pull request merged**